snuba/datasets/transactions_processor.py

```diff
@@ -229,4 +291,28 @@ def process_message(
         if processed["sdk_version"] == "":
             metrics.increment("missing_sdk_version")
 
+    def process_message(
+        self, message: Tuple[int, str, Dict[Any, Any]], metadata: KafkaMessageMetadata
+    ) -> Optional[ProcessedMessage]:
+        structured_message = self._structure_and_validate_message(message)
+        if not structured_message:
+            return None
+        processed: MutableMapping[str, Any] = {
+            "deleted": 0,
+            "retention_days": structured_message.retention_days,
+        }
+        # The following helper functions should be able to be applied in any order.
+        # At time of writing, there are no reads of the values in the `processed`
+        # dictionary to inform values in other functions.
+        # Ideally we keep continue that rule
+        self._process_base_event_values(processed, structured_message)
```

Comment:
See my comment above. I don't think `StructuredMessage` is necessary since the only fields that get used on it are `retention_days` and `event` (`data` is a field of `event`). I would return those two values from the validate function and then pass the `event` or `event["data"]` as appropriate into the helper functions.